finesse/gui/led_icons.py

```diff
@@ -5,38 +5,47 @@
 from PySide6.QtGui import QImage, QPixmap
 from PySide6.QtWidgets import QLabel
 
-img_files = resources.files("finesse.gui.images")
-poll_on_img_data = img_files.joinpath("poll_on.png").read_bytes()
-poll_off_img_data = img_files.joinpath("poll_off.png").read_bytes()
-alarm_on_img_data = img_files.joinpath("alarm_on.png").read_bytes()
-alarm_off_img_data = img_files.joinpath("alarm_off.png").read_bytes()
+_img_files = resources.files("finesse.gui.images")
+_poll_on_img_data = _img_files.joinpath("poll_on.png").read_bytes()
+_poll_off_img_data = _img_files.joinpath("poll_off.png").read_bytes()
+_alarm_on_img_data = _img_files.joinpath("alarm_on.png").read_bytes()
+_alarm_off_img_data = _img_files.joinpath("alarm_off.png").read_bytes()
 
-poll_on_img = QImage.fromData(poll_on_img_data)
-poll_off_img = QImage.fromData(poll_off_img_data)
-alarm_on_img = QImage.fromData(alarm_on_img_data)
-alarm_off_img = QImage.fromData(alarm_off_img_data)
+_poll_on_img = QImage.fromData(_poll_on_img_data)
+_poll_off_img = QImage.fromData(_poll_off_img_data)
+_alarm_on_img = QImage.fromData(_alarm_on_img_data)
+_alarm_off_img = QImage.fromData(_alarm_off_img_data)
 
 
 class LEDIcon(QLabel):
     """QLabel object to represent an LED with on/off status."""
 
-    def __init__(self, status: int) -> None:
-        """Creates the LED icon, sets its status and stores corresponding image data."""
+    def __init__(self, on_img: QImage, off_img: QImage, is_on: bool = False) -> None:
+        """Creates the LED icon, sets its status and stores corresponding image data.
+
+        Args:
+            on_img (QImage): QImage for LED on state.
+            off_img (QImage): QImage for LED off state.
+            is_on (bool): On/off status of LED.
+        """
         super().__init__()
-        self._status = status
-        self._on_img = QImage()
-        self._off_img = QImage()
+        self._on_img = QImage(on_img)
+        self._off_img = QImage(off_img)
+        if is_on:
+            self._turn_on()
+        else:
+            self._turn_off()
         self._timer = QTimer()
         self._timer.timeout.connect(self._turn_off)  # type: ignore
 
     def _turn_on(self):
         """Turns the LED on."""
-        self._status = 1
+        self._is_on = True
         self.setPixmap(QPixmap(self._on_img))
 
     def _turn_off(self):
         """Turns the LED off."""
-        self._status = 0
+        self._is_on = 0
```

Comment:
```suggestion
        self._is_on = False
```

Reply:
> Found a couple of really minor things.

Thanks! I should have been more careful with converting int to bool!

> Something else I probably should have asked earlier is, where did you get these icons from? What licence (if any) are they released under?

Good question - I made them on Inkscape haha so fully free and open source!